scipy/io/matlab/mio5_utils.pyx

```diff
@@ -497,15 +503,22 @@ cdef class VarReader5:
            Number of integers read
         '''
         cdef:
-            cnp.uint32_t mdtype, byte_count
-            int i
+            cnp.uint32_t mdtype, byte_count, n_ints
+            int i, check_ints=0
         self.read_element_into(&mdtype, &byte_count, <void *>int32p)
-        if mdtype != miINT32:
+        if mdtype == miUINT32:
+            check_ints = 1
+        elif mdtype != miINT32:
             raise TypeError('Expecting miINT32 as data type')
-        cdef int n_ints = byte_count // 4
+        n_ints = byte_count // 4
         if self.is_swapped:
             for i in range(n_ints):
                 int32p[i] = byteswap_u4(int32p[i])
+        if check_ints:
+            for i in range(n_ints):
+                if int32p[i] < 0:
+                    raise ValueError('Expecting miINT32, got miUINT32 with '
+                                     'negative values')
```

Comment:
Yes, it's due to the fact that there appear to be buggy matlab files (not written by matlab itself) that note the data as UINT32 rather than the standard INT32 - the check is just to make sure that the values didn't overflow.